agent/config/builder.go

```diff
@@ -1918,6 +1927,136 @@ func (b *Builder) isUnixAddr(a net.Addr) bool {
+	var val AutoConfigAuthorizer
+
+	val.Enabled = b.boolValWithDefault(raw.Enabled, false)
+	val.ClaimAssertions = raw.ClaimAssertions
+	val.AllowReuse = b.boolValWithDefault(raw.AllowReuse, false)
+	val.AuthMethod = structs.ACLAuthMethod{
+		Name: "Auto Config Authorizer",
+		Type: "jwt",
+		// TODO (autoconf) - Configurable token TTL
+		MaxTokenTTL:    72 * time.Hour,
+		EnterpriseMeta: *structs.DefaultEnterpriseMeta(),
+		Config: map[string]interface{}{
+			"JWTSupportedAlgs":     raw.JWTSupportedAlgs,
+			"BoundAudiences":       raw.BoundAudiences,
+			"ClaimMappings":        raw.ClaimMappings,
+			"ListClaimMappings":    raw.ListClaimMappings,
+			"OIDCDiscoveryURL":     b.stringVal(raw.OIDCDiscoveryURL),
+			"OIDCDiscoveryCACert":  b.stringVal(raw.OIDCDiscoveryCACert),
+			"JWKSURL":              b.stringVal(raw.JWKSURL),
+			"JWKSCACert":           b.stringVal(raw.JWKSCACert),
+			"JWTValidationPubKeys": raw.JWTValidationPubKeys,
+			"BoundIssuer":          b.stringVal(raw.BoundIssuer),
+			"ExpirationLeeway":     b.durationVal("auto_config.authorizer.expiration_leeway", raw.ExpirationLeeway),
+			"NotBeforeLeeway":      b.durationVal("auto_config.authorizer.not_before_leeway", raw.NotBeforeLeeway),
+			"ClockSkewLeeway":      b.durationVal("auto_config.authorizer.clock_skew_leeway", raw.ClockSkewLeeway),
+		},
+		// should be unnecessary as we aren't using the typical login process to create tokens but this is our
+		// desired mode regardless so if it ever did matter its probably better to be explicit.
+		TokenLocality: "local",
+	}
+
+	return val
+}
+
+func (b *Builder) validateAutoConfig(rt RuntimeConfig) error {
+	autoconf := rt.AutoConfig
+
+	if autoconf.Enabled {
+		// Auto Config doesn't currently support configuring servers
+		if rt.ServerMode {
+			return fmt.Errorf("auto_config.enabled cannot be set to true for server agents.")
+		}
+
+		// When both are set we will prefer the given value over the file.
+		if autoconf.IntroToken != "" && autoconf.IntroTokenFile != "" {
+			b.warn("auto_config.intro_token and auto_config.intro_token_file are both set. Using the value of auto_config.intro_token")
+		} else if autoconf.IntroToken == "" && autoconf.IntroTokenFile == "" {
+			return fmt.Errorf("one of auto_config.intro_token or auto_config.intro_token_file must be set to enable auto_config")
+		}
+
+		if len(autoconf.ServerAddresses) == 0 {
+			// TODO (autoconf) can we/should we infer this from the join/retry join addresses. I think no, as we will potentially
+			// be overriding those retry join addresses with the autoconf process anyways.
+			return fmt.Errorf("auto_config.enabled is set without providing a list of addresses")
+		}
+
+		// TODO (autoconf) should we validate the DNS and IP SANs? The IP SANs have already been parsed into IPs
+	}
+
+	return b.validateAutoConfigAuthorizer(rt)
+}
+
+func (b *Builder) validateAutoConfigAuthorizer(rt RuntimeConfig) error {
+	authz := rt.AutoConfig.Authorizer
+
+	if authz.Enabled {
```

Comment:
Same thought about the guard and warning

Reply:
Done.